File: library-net/src/main/java/com/my/library_net/interceptor/BasicDataInterceptor.java
package com.my.library_net.interceptor;

import java.io.IOException;
import java.net.URLDecoder;

import okhttp3.HttpUrl;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.RequestBody;
import okhttp3.Response;
import okio.Buffer;

public class BasicDataInterceptor implements Interceptor {
    private static final String METHOD_GET = "GET";
    private static final String METHOD_POST = "POST";


    @Override
    public Response intercept(Interceptor.Chain chain) throws IOException {
        Request originRequest = chain.request();
        HttpUrl httpUrl = originRequest.url();
        StringBuilder requestUrl = new StringBuilder();
        requestUrl.append(httpUrl.url().toString() + "?");
        try {
            // GET方法
            if (METHOD_GET.equals(originRequest.method())) {
                requestUrl.append(httpUrl.toString());

            } else if (METHOD_POST.equals(originRequest.method())) {
                RequestBody requestBody = originRequest.body();
                if (requestBody != null) {
                    String msg = "-url--" + originRequest.url();
                    if (msg.contains("uploadFile")) {
                        requestUrl.append("--上传文件内容--");
                    } else {
                        requestUrl.append(getParam(requestBody));
                    }

                }
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
        return chain.proceed(originRequest);
    }


    /**
     * 读取参数
     *
     * @param requestBody
     * @return
     */
    private String getParam(RequestBody requestBody) {
        Buffer buffer = new Buffer();
        String logparm;
        try {
            requestBody.writeTo(buffer);
            logparm = buffer.readUtf8();
            logparm = URLDecoder.decode(logparm, "utf-8");
        } catch (IOException e) {
            e.printStackTrace();
            return "";
        }
        return logparm;
    }
}

File: library-net/src/main/java/com/my/library_net/interceptor/BasicDataInterceptor.kt
package com.my.library_net.interceptor

import okhttp3.*
import okio.Buffer
import java.io.IOException
import java.net.URLDecoder

class BasicDataInterceptor constructor() : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val originRequest: Request = chain.request()
        val httpUrl: HttpUrl = originRequest.url()
        val requestUrl: StringBuilder = StringBuilder()
        requestUrl.append(httpUrl.url().toString() + "?")
        try {
            // GET方法
            if ((METHOD_GET == originRequest.method())) {
                requestUrl.append(httpUrl.toString())
            } else if ((METHOD_POST == originRequest.method())) {
                val requestBody: RequestBody? = originRequest.body()
                if (requestBody != null) {
                    val msg: String = "-url--" + originRequest.url()
                    if (msg.contains("uploadFile")) {
                        requestUrl.append("--上传文件内容--")
                    } else {
                        requestUrl.append(getParam(requestBody))
                    }
                }
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return chain.proceed(originRequest)
    }

    /**
     * 读取参数
     *
     * @param requestBody
     * @return
     */
    private fun getParam(requestBody: RequestBody): String {
        val buffer: Buffer = Buffer()
        var logparm: String
        try {
            requestBody.writeTo(buffer)
            logparm = buffer.readUtf8()
            logparm = URLDecoder.decode(logparm, "utf-8")
        } catch (e: IOException) {
            e.printStackTrace()
            return ""
        }
        return logparm
    }

    companion object {
        private const val METHOD_GET: String = "GET"
        private const val METHOD_POST: String = "POST"
    }
}